### ISSUE TITLE
Add try operator

### ISSUE THREAD

**Issue by kRITZCREEK** [purescript/purescript-exceptions#15]
Could we add a try operator like Control.Exception has in Haskell? I'm currently using this:

```purescript
try ∷ ∀ a e. Eff (err ∷ EXCEPTION | e) a → Eff e (Either Error a)
try action = catchException (pure ∘ Left) (Right <$> action)

```

The only problem would be that this introduces a dependency on `purescript-either`.

**Comment by garyb** [comment]
I don't think having that dependency would be a problem. What say you @paf31?

**Comment by paf31** [comment]
Sounds good to me.

**Comment by kRITZCREEK** [comment]
Fixed with #16
